api/src/main/java/com/graphhopper/client/AbstractHopperClient.java

```diff
@@ -0,0 +1,122 @@
+package com.graphhopper.client;
+
+
+import com.graphhopper.GHResponse;
+import com.graphhopper.HopperClient;
+import com.graphhopper.HopperRequest;
+import com.graphhopper.HopperResponse;
+import com.graphhopper.bean.RouteBbox;
+import com.graphhopper.bean.RouteInstruction;
+import com.graphhopper.http.GraphHopperWeb;
+import com.graphhopper.http.WebHelper;
+import com.graphhopper.internal.HopperEngine;
+import com.graphhopper.internal.LazyPointList;
+import com.graphhopper.util.*;
+import com.graphhopper.util.shapes.BBox;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+// Please do not throw an error if hasError (with check everywhere)
+// Throws errors on errors is an anti-pattern
```

Comment:
If you want to enforce the hasError check then make clear examples\documentation with it. No need to waste CPU clock at runtime in my opinion.